### ISSUE TITLE
Expose data_rw_flags to the user

### ISSUE THREAD

**Issue by philipheinisch** [IANW-Projects/toolkitICL#7]
We should expose data_rw_flags to the user, to be able to specify if some varibales are read/write only or both. This could be extended to support different memory copy options for multiple kernels in the future.

**Comment by ranocha** [comment]
Cf. https://github.com/IANW-Projects/toolkitICL/blob/07b12453b66bbdbdf6c2513e44d5c33d6ee97b8a/src/main.cpp#L236

**Comment by philipheinisch** [comment]
Exactly, this just sets everything to read & write. We should at least expand the 'TODO' to explain what data_rw_flags does or can do.
